Sources/Workspace/Workspace.swift

```diff
@@ -941,6 +947,17 @@ public class Workspace {
                 case .checkout(let checkoutState):
                     packagePath = checkoutsPath.appending(managedDependency.subpath)
                     version = checkoutState.version
+                    
+                    // In case the checkouts folder for the dependency was deleted,
+                    // clone it again.
+                    if !fileSystem.isDirectory(packagePath) {
```

Comment:
We may also need a method which computes the path of a managed dependency given checkout and editables directories. This method can then be used here and in the new validate method.